R/html-tweak.R

```diff
@@ -78,13 +78,7 @@ tweak_all_links <- function(html, pkg = pkg) {
 
   hrefs <- xml2::xml_attr(links, "href")
   # Users might have added absolute URLs to e.g. the Code of Conduct
-  if (is.null(pkg$meta$url)) {
-    needs_tweak <- grepl("https?\\:\\/\\/", hrefs)
-  } else {
-    url <- pkg$meta$url
-    needs_tweak <- grepl("https?\\:\\/\\/", hrefs) & !grepl(url, hrefs)
-  }
-  tweak_class_prepend(links[needs_tweak], "external-link")
+  tweak_class_prepend(links[!is_internal_link(hrefs, pkg = pkg)], "external-link")
```

Comment:
this is another place where we needed to judge whether an URL was an external URL.